module/lua/ldo.c

```diff
@@ -85,6 +86,11 @@ void longjmp (label_t * buf) {
 	for (;;);
 }
 #endif
+#else
+#define LUAI_THROW(L,c)		longjmp((c)->b, 1)
+#define LUAI_TRY(L,c,a)		if (setjmp((c)->b) == 0) { a }
+#define luai_jmpbuf		jmp_buf
```

Comment:
we have different defaults for _KERNEL and userland on dilos:
https://bitbucket.org/dilos/dilos-illumos/src/16dd6d2ada392d7b6231ef5b2c638983aef6906b/usr/src/uts/common/fs/zfs/lua/ldo.c#lines-49

in this context - what is it default we can identify?

Reply:
I'm testing a different patch to hopefully solve this.

Reply:
@ikozhukhov This adds a case for kernel’s that support user space semantics. In your case I’d just expect you to add an `#else if __dilos__` with the snippet supporting your needs.

Reply:
well, it'll be copy-past from from our file - will it be applicable?
yes, we have `__dilos__` and `__dilos` and use it with our platform builds

Reply:
@ikozhukhov if it’s just a subset of the linux code, can’t you just do `#if defined(__linux__) || defined(__dilos__)` ?

Reply:
need investigate it because we have addition check with comments: https://bitbucket.org/dilos/dilos-illumos/src/16dd6d2ada392d7b6231ef5b2c638983aef6906b/usr/src/uts/common/fs/zfs/lua/ldo.c#lines-42

Reply:
Let's do as @mattmacy suggests when the time comes.

Reply:
i'm not sure it's correct for multiple platforms - probably have to move part of changes to separate platform depend file and include it to shared file

Reply:
These changes aren't set in stone. When you start to do diff reduction on your end we can revisit this.

Reply:
well, will not stop you in your drive :)
go ahead with your needs in this time, i'll take a look how i can reuse/update it to dilos needs.